rasa/shared/core/domain.py

```diff
@@ -1916,6 +1919,66 @@ def _collect_actions_which_explicitly_need_domain(
 
         return action_names
 
+    @staticmethod
+    def _collect_response_ids(
+        responses: Dict[Text, List[Dict[Text, Any]]]
+    ) -> Dict[Text, Set[Text]]:
+        """Collects all response ids.
+
+        Args:
+            responses: The responses to collect the ids from.
+
+        Returns:
+            A dictionary mapping the response names to the set of response ids.
+        """
+        response_ids = set()
+        response_ids_per_response: Dict[Text, Set[Text]] = {}
+        for response_name, response_variations in responses.items():
+            response_ids_for_response = (
+                Domain._collect_response_ids_for_response_variations(
+                    response_variations
+                )
+            )
+
+            already_present_response_ids = response_ids_for_response.intersection(
+                response_ids
+            )
+
+            if len(already_present_response_ids) > 0:
+                raise RasaException(
+                    f"Duplicate response ids "
+                    f"'{already_present_response_ids}' "
+                    f"defined in domain."
+                )
```

Comment:
Should we raise a user warning and continue with loading the Domain regardless? We should consider including this check actually in rasa data validate which also run in `rasa train`. This way it can be detected early and prevent the application from crashing prematurely.